dash/App/Chains/Connection/index.js

```diff
@@ -78,20 +108,21 @@ class ChainModule extends React.Component {
       </div>
     )
   }
+
   status (type, id, layer) {
     const connection = this.store('main.networks', type, id, 'connection', layer)
     let status = connection.status
     const current = connection.current
 
     if (current === 'custom') {
       if (layer === 'primary' && this.state.primaryCustom !== '' && this.state.primaryCustom !== this.customMessage) {
-        if (!this.okProtocol(this.state.primaryCustom)) status = 'invalid target'
-        else if (!this.okPort(this.state.primaryCustom)) status = 'invalid port'
+        if (!okProtocol(this.state.primaryCustom)) status = 'invalid target'
```

Comment:
can we do these validations before setting the target as well? I see a bunch of issues being caused, including the unhandled `chain` exceptions because we display the error to the user but still set the target to whatever was typed

Reply:
also lets add some presets so not let users connect back to Frame itself (`ws/httplocalhost:1248 127.0.0.1:1248`)